pkg/sfu/session.go

```diff
@@ -189,3 +195,35 @@ func (s *Session) Peers() map[string]*Peer {
 func (s *Session) OnClose(f func()) {
 	s.onCloseHandler = f
 }
+
+func (s *Session) audioLevelObserver(audioLevelInterval int) {
+	if audioLevelInterval < 500 {
+		audioLevelInterval = 500
+	}
+	if audioLevelInterval > 5000 {
+		audioLevelInterval = 5000
+	}
+	for {
+		time.Sleep(time.Duration(audioLevelInterval) * time.Millisecond)
+		if s.closed.get() {
+			return
+		}
+		levels := s.audioObserver.calc()
+
+		s.mu.RLock()
+		defer s.mu.RUnlock()
+		l, err := json.Marshal(&levels)
+		if err != nil {
+			log.Errorf("Marshaling audio levels err: %v", err)
+		}
+		sl := string(l)
+
+		for _, peer := range s.peers {
+			if ch, ok := peer.subscriber.channels[APIChannelLabel]; ok {
+				if err = ch.SendText(sl); err != nil {
+					log.Errorf("Sending audio levels to peer: %s, err: %v", peer.id, err)
+				}
+			}
+		}
+	}
+}
```

Comment:
There's more needed here right? At the moment it sends an array of stream ids, but there's no way for the client to know which ones are making noise.

Reply:
there is only returned streams with noise ordered:

```golang
	if dBov <= a.threshold {
				as.sum += int(dBov)
				as.total++
			}
```